bgpd/bgp_route.c

```diff
@@ -2903,6 +2913,17 @@ int bgp_update(struct peer *peer, struct prefix *p, uint32_t addpath_id,
 	else
 		has_valid_label = bgp_is_valid_label(label);
 
+	/* If the route is newly learnt from peer and suppress fib is
+	 * configured, advertise route when route is installed in fib
+	 */
+	if (bgp && CHECK_FLAG(bgp->flags, BGP_FLAG_SUPPRESS_FIB_PENDING)
+		&& (sub_type == BGP_ROUTE_NORMAL)) {
+		if (rn->info == NULL)
+			SET_FLAG(rn->flags, BGP_NODE_FIB_INSTALL_PENDING);
```

Comment:
This flag should never be set in the case where bgpd is operating without zebra needs to be covered -- i.e., bgp_zebra_num_connect()==0 || bgp_option_check(BGP_OPT_NO_FIB)